backend/core/rule/tree.py

```diff
@@ -120,7 +234,31 @@ def find(self, name):
 
 class TreeNode(object):
     def __init__(self, data, properties, metadata, func=None, namespace=None, *args):
+        """Constructor of TreeNode
+
+        Args:
+            data: data of current node. more additional explain is in below
+            properties: dict of property. will be deconstructed
+            metadata: data of any other information needed evaluating tree
+            func: evaluator function (closure)
```

Comment:
nope, closure is right

closure's meaning can be referenced in [here](https://en.wikipedia.org/wiki/Closure_(computer_programming))